### ISSUE TITLE
Improper application of rad2deg in transform_points

### ISSUE THREAD

**Issue by Jeitan** [SciTools/cartopy#623]
I have been doing some comparisons between cartopy and basemap to determine which one I want to use, and I ran into the issue that cartopy had an apparent transform error when converting from basic x, y, z geocentric coordinates to geodetic coordinates, and vice versa.

This snippet illustrates the issue using basic pyproj and using cartopy:

```python
from mpl_toolkits.basemap import pyproj

x, y, z = 2574.32516e3, 837.562e3, 5761.325e3

gc = pyproj.Proj(r"+proj=geocent")
gd = pyproj.Proj(r"+proj=lonlat")
lon, lat, alt = pyproj.transform(gc, gd, x, y, z)
print(lon, lat, alt)

import cartopy.crs as ccrs
import numpy as np
cgc = ccrs.Geocentric()
cgd = ccrs.Geodetic()
output = cgd.transform_points(cgc, np.array(x),
                                   np.array(y),
                                   np.array(z))
print(output[0][0], output[0][1], output[0][2])
```
The result is

    18.0224043189 64.9796515089 5048.03893734
    18.0224043189 64.9796515089 289231.325927

The first line is correct, the second is obviously not.  Of course this seemed odd because both are based on PROJ.4.  I tracked the issue to lines 345 and 346 in _crs.pyx, in the definition of transform_points:

        if self.is_geodetic():
            result = np.rad2deg(result)

The result at this point is lat, lon, height in meters, and performing rad2deg on meters is just plain wrong.  This is confirmed by `np.deg2rad(289231.325927) = 5048.0389373`

The reverse transform is also affected by lines 338 and 339; in order to get the right answer one must insert `np.rad2deg(altitude in meters)`:

        if src_crs.is_geodetic():
            result = np.deg2rad(result)

I'm not nearly familiar enough with the innards of cartopy to suggest a fix, so I'll leave it to the experts!

**Comment by rhattersley** [comment]
Thanks for tracking this issue down and isolating it so clearly. :+1: :smile:

**Comment by pelson** [comment]
:+1: - great investigation @Jeitan. Thanks

**Comment by QuLogic** [comment]
@pelson This change needs to be backported from master if you want it on 0.12.1.

**Comment by pelson** [comment]
> This change needs to be backported from master if you want it on 0.12.1.

Good point. Maybe we should just move forwards and call it 0.13.0.

**Comment by pelson** [comment]
Fixed by #625.
